src/lua/LuaEngine.cpp

```diff
@@ -1082,6 +1092,329 @@ static int l_engine_get_sector_map_factions(lua_State *l)
 	return 1;
 }
 
+static int l_engine_system_map_set_color(lua_State *l)
+{
+	SystemView *sv = Pi::game->GetSystemView();
```

Comment:
Please prefer making `SystemMapSetColor` and its ilk member functions on a new `LuaObject<SystemView>` class instead, and add the accessor for the instance of the SystemView to the `LuaGame` interface. I'd much prefer that we don't start C-style namespacing functions and putting them in `LuaEngine` when they don't actually need to be there.

Yes, I know that's how the SectorMap does it right now, but it really shouldn't be done that way.

Reply:
Made in 7dfc75ba29538d8d142bc7b88e500906ff2ad93c